zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/UnifiedServerSocket.java

```diff
@@ -255,7 +255,7 @@ private void detectMode() throws IOException {
                 bytesRead = 0;
             }
 
-            if (bytesRead == litmus.length && SslHandler.isEncrypted(Unpooled.wrappedBuffer(litmus))) {
+            if (bytesRead == litmus.length && SslHandler.isEncrypted(Unpooled.wrappedBuffer(litmus), false)) {
```

Comment:
This variant is brand new in `netty-4.1.113.Final`, so this patch will enforce netty version. Given my comments https://github.com/apache/zookeeper/pull/1917#issuecomment-2345528254, I am +1 to this. So, client will have to bump its netty version to match this patch in compilation.

netty commit: https://github.com/netty/netty/commit/dc30c3337a766d9f9519e910438ed3a3ae1b3761#diff-963dd9b8e77e913395273812889e9b6f2f449fe55e2b748b95311b219d1c58faR1286

Reply:
Isn't that a problem if we backport this to 3.9? (I want to)
Can we just use another variant which is backward compatible?

Reply:
The only other variant is the deprecated one, which fails to compile due to the `-Werror` flag. This file change could possibly be left out of a backport which uses an older netty version.

Reply:
No worries, we'll upgrade on 3.9 too.